src/auth/tests/credentials.rs

```diff
@@ -22,24 +22,22 @@ mod test {
 
     #[tokio::test]
     #[serial_test::serial]
-    async fn create_access_token_credential_no_adc_filepath_fallback() {
-        // TODO(#442) - We should (successfully) fall back to MDS Credentials. But for now we are temporarily returning an error.
+    async fn create_access_token_credential_no_adc_filepath_fallback_to_mds() {
         let _e1 = ScopedEnv::remove("GOOGLE_APPLICATION_CREDENTIALS");
         let _e2 = ScopedEnv::remove("HOME"); // For posix
         let _e3 = ScopedEnv::remove("APPDATA"); // For windows
-        let err = create_access_token_credential().await.err().unwrap();
-        let msg = err.source().unwrap().to_string();
-        assert!(msg.contains("Unable to find Application Default Credentials"));
+        create_access_token_credential().await.unwrap();
     }
 
     #[tokio::test]
     #[serial_test::serial]
-    async fn create_access_token_credential_no_adc_file_fallback() {
-        // TODO(#442) - We should (successfully) fall back to MDS Credentials. But for now we are temporarily returning an error.
+    async fn create_access_token_credential_errors_if_adc_env_is_not_a_file() {
```

Comment:
Seems like you should be able to add a test like:

```rust
#[tokio::test]
#[serial_test::serial]
async fn create_access_token_credential_no_well_known_adc() -> std::result::Result<(), Box<dyn std::error::Error>>  {
  let dir = testfile::tempdir()?;
  let _e1 = ScopedEnv::remove("GOOGLE_APPLICATION_CREDENTIALS");
  let _e2 = ScopedEnv::set("HOME", format!("{}/blah blah", tempdir.path()); 
  let _e3 = ScopedEnv::set("APPDATA", format!({}/blah blah", tempdir.path());

  let e = create_access_token_credential().await;
  assert!(e.is_err());

  Ok(())
}
```

Reply:
Yeah, that is clever. I did not think of overriding `HOME` / `APPDATA`. Added a test. Thanks.